.teamcity/_self/projects/WebApp.kt

```diff
@@ -575,19 +575,61 @@ object CheckCodeStyleBranch : BuildType({
 
 				# Find files to lint
 				TOTAL_FILES_TO_LINT=$(git diff --name-only --diff-filter=d refs/remotes/origin/trunk...HEAD | grep -cE '(\.[jt]sx?|\.json|\.md)${'$'}' || true)
+				FILES_TO_LINT=$(git diff --name-only --diff-filter=d refs/remotes/origin/trunk...HEAD | grep -E '(\.[jt]sx?|\.json|\.md)${'$'}' || echo "")
 
-				# Avoid running more than 16 parallel eslint tasks as it could OOM
-				if [ "%run_full_eslint%" = "true" ] || [ "${'$'}TOTAL_FILES_TO_LINT" -gt 16 ] || [ "${'$'}TOTAL_FILES_TO_LINT" == "0" ]; then
+				# Create temporary output directory
+				mkdir -p "./checkstyle_results/eslint"
+
+				if [ "%run_full_eslint%" = "true" ] || [ "${'$'}TOTAL_FILES_TO_LINT" == "0" ]; then
```

Comment:
I'm erring on the side of caution here, because `== "0"` actually behaves the same way. The reason why I'm suggesting this change is because it could lead novice Bash users to think that number comparisons are possible with `== != > <` etc., whereas `-eq -ne -gt -lt` should be used.

```suggestion
				if [ "%run_full_eslint%" = true ] || [ "${'$'}TOTAL_FILES_TO_LINT" -eq 0 ]; then
```

Reply:
Second question: I'm having a hard time finding a good reference on the syntax expected by TeamCity inside `scriptContent`. Although I don't like asking LLMs for this sort of information, one model suggested that dollar signs can be escaped via simple repetition: `$$TOTAL_FILES_TO_LINT`. This wouldn't surprise me, since it's a convention that Makefiles started 48 years ago. :)

If that's the case and not hallucination, we should clean up this whole script.

Reply:
> Second question: I'm having a hard time finding a good reference on the syntax expected by TeamCity inside scriptContent

We know the answer to this now :) https://github.com/Automattic/wp-calypso/pull/100986#issuecomment-2714913974